chrome/content/zotero/xpcom/zotero.js

```diff
@@ -507,6 +509,35 @@ Components.utils.import("resource://gre/modules/osfile.jsm");
 			}
 		}
 		
+		// Check for DB purge
+		var newAccountFile = dataDir.clone();
+		newAccountFile.append('new-account');
+		if (newAccountFile.exists()) {
+			try {
+				// TODO: better error handling?
+				Zotero.debug("New account file detected. Removing all files", 1);
+				
+				var storageDir = dataDir.clone();
+				storageDir.append('storage');
+				if (storageDir.exists()) {
+					storageDir.remove(true);
+				}
+				
+				var dbfile = Zotero.getZoteroDatabase();
+				if (dbfile.exists()) {
+					dbfile.remove(false);
+				}
+				
+				newAccountFile.remove(false);
+
+				Zotero.newAccount = true;
+			}
+			catch (e) {
+				// Restore from backup?
+				alert(e);
+			}
+		}
```

Comment:
We should just combine this with the restore-from-server stuff above, which can also delete 'storage'.